synapse/lib/view.py

```diff
@@ -824,6 +824,56 @@ async def getPropArrayCount(self, propname, valu=s_common.novalu):
 
         return count
 
+    async def iterPropUniqValues(self, propname):
+        prop = self.core.model.prop(propname)
+        if prop is None:
+            mesg = f'No property named {propname}'
+            raise s_exc.NoSuchProp(mesg=mesg)
+
+        formname = None
+        propname = None
+
+        if prop.isform:
+            formname = prop.name
+        else:
+            propname = prop.name
+            if not prop.isuniv:
+                formname = prop.form.name
+
+        async def wrapgenr(lidx, genr):
+            async for indx, valu in genr:
+                yield indx, valu, lidx
+
+        genrs = []
+        for lidx, layr in enumerate(self.layers):
+            genr = layr.iterPropUniqValues(formname, propname, prop.type.stortype)
+            genrs.append(wrapgenr(lidx, genr))
+
+        lastvalu = None
+        async for indx, valu, lidx in s_common.merggenr2(genrs):
+            if valu == lastvalu:
+                continue
+
+            if lidx == 0 or propname is None:
+                lastvalu = valu
+                yield valu
+            else:
+                valid = False
+                async for buid in self.layers[lidx].iterPropIndxBuids(formname, propname, indx):
```

Comment:
There are some index values that we can use to reconstruct the property value without needing to lift the sode. Any chance we could use that when possible?

Reply:
We use that in `iterPropUniqValues` on the layer but in views with multiple layers we need to check if the property is overwritten with a different value in a higher layer.

Reply:
If it is value from the top layer we known it is valid for the view, if it is from a lower layer, we iterate the buids for that prop value to find at least one instance where that value isn't overwritten in a layer above it.